### PR TITLE
Restrict workplan tab actions

### PR DESCRIPTION
**Pull request by jadmnsaad**
https://app.zenhub.com/workspaces/epictrack-63891ea941d309001fa292cf/issues/gh/bcgov/epic.track/1579

Details:
- Use Restrict component to wrap buttons
- Add auth_check for all the services in the backend
- Add work_id in the request body to some bulk operations to be able to use it in auth_check

### CODE REVIEW

File: epictrack-api/src/api/resources/event.py [hunk 1]
```diff
@@ -100,22 +99,6 @@ def delete(event_id):
         return message, HTTPStatus.OK
 
 
-@cors_preflight("DELETE")
-@API.route("/events", methods=["DELETE", "OPTIONS"])
-class MilestoneEvents(Resource):
-    """Endpoint resource to manage milestone events"""
-
-    @staticmethod
-    @cors.crossdomain(origin="*")
```
Comment (dinesh-aot) [inline epictrack-api/src/api/resources/event.py]: Are you deleting this because we are not using it at all?

Comment (jadmnsaad) [reply epictrack-api/src/api/resources/event.py]: yup, no calls for it in the UI

File: epictrack-api/src/api/services/task.py [hunk 2]
```diff
@@ -241,8 +271,14 @@ def bulk_update(cls, data: dict):
         return "Updated successfully"
 
     @classmethod
-    def bulk_delete_tasks(cls, task_ids: List):
+    def bulk_delete_tasks(cls, task_ids: List, work_id: int):
```
Comment (jadmnsaad) [inline epictrack-api/src/api/services/task.py:274]: @dinesh-aot @saravanpa-aot should we add a "Delete" role for operations like this. We only have Create and Edit now

Comment (dinesh-aot) [reply epictrack-api/src/api/services/task.py:274]: Good Question. I think we should create a role called Delete. @saravanpa-aot thoughts?

Comment (jadmnsaad) [reply epictrack-api/src/api/services/task.py:274]: Added this ticket for the Delete role https://app.zenhub.com/workspaces/epictrack-63891ea941d309001fa292cf/issues/gh/bcgov/epic.track/1679, happy to merge this once approved

File: epictrack-api/src/api/resources/event.py (hunk 1 above)
Comment (dinesh-aot) [reply epictrack-api/src/api/resources/event.py]: @jadmsaadaot Lets keep this method cz we might call it from UI in future for some other roles or we might use this api from postman to get stuff done at some point. So better we keep it.

**Comment by jadmnsaad** [comment]
image
pylint is complaining that the module services/event.py is too big, exceeds lines of code limit. Fixing that would some refactoring and is really out of scope of this, can we ignore it?

**Comment by saravanpa-aot** [comment]
> image pylint is complaining that the module services/event.py is too big, exceeds lines of code limit. Fixing that would some refactoring and is really out of scope of this, can we ignore it?

sure... plz create a ticket for refactoring..put the above screenshot

**Comment by jadmnsaad** [comment]
> > image pylint is complaining that the module services/event.py is too big, exceeds lines of code limit. Fixing that would some refactoring and is really out of scope of this, can we ignore it?
> 
> sure... plz create a ticket for refactoring..put the above screenshot

added https://app.zenhub.com/workspaces/epictrack-63891ea941d309001fa292cf/issues/gh/bcgov/epic.track/1681

**Pull request merged**